src/preact.d.ts

```diff
@@ -141,6 +147,7 @@ declare namespace JSX {
 		y1?:number | string;
 		y2?:number | string;
 		y?:number | string;
+		[propName: string]: any;
```

Comment:
One more thing I added was this catchall so we can have `data-*` attributes on HTML elements